libs/ui/src/lib/hooks/use-toggle.tsx

```diff
@@ -0,0 +1,21 @@
+import {Dispatch, SetStateAction, useCallback, useState} from 'react'
+
+export type UseToggle = {
```

Comment:
No need to type it really, hooks usually infer very well due to the rule of hooks.